### DISCUSSION TITLE
a/macos-deploy-immich

### DISCUSSION THREAD

**Discussion by giscus** [sincerefly/giscus-data#14] [Announcements]
# a/macos-deploy-immich

https://blog.yasking.org/a/macos-deploy-immich.html

**Comment by qqbgkiki** [comment]
你好，按你的操作一步一步走，“修改 .env 环境变量”  这一步是不是少了终端命令呢，我自己百度执行了不知道对不。
然后到“编辑 docker-compose.yml 配置文件”这一步，我添加后启动不了，不知道是哪里问题。我是完全小白 一步一步都是边看边百度的。

**Reply by sincerefly** [reply]
你好，文章已更新，感谢反馈问题。

> “修改 .env 环境变量” 可以使用 vim 或者 VS Code 编辑器编辑，在文章中做了更详尽的补充
> “编辑 docker-compose.yml 配置文件” 添加后启动不了大概率是配置文件内容的格式改坏了，参考我补充的截图修改试试

Immich 开发阶段变总会变更，推荐“硬看”官方文档结合 Issues 和 AI 解决遇到的问题，如需进一步帮助也欢迎给我发送邮件

**Comment by BianTan** [comment]
用来跑里面的机器学习快吗

**Reply by sincerefly** [reply]
算挺快的吧，m2 跑起来没什么压力

**Comment by zm995** [comment]
感谢分享，小白希望能出个视频教程跟着一帧一帧做^_^

**Reply by sincerefly** [reply]
如果是“真小白”不太建议折腾 immich，因为后续也会涉及到维护管理的问题，不如观望几年，等其稳定后，可能就有基于 immich 的开箱即用的产品出来了
如果想折腾体验，可以先了解命令行、Docker、Docker Compose、Docker 换源（如果在国内）相关的概念和知识，再对照本文或网络上搜索最新的部署文章，就能“无门槛”部署了 😄

**Comment by zhz-we** [comment]
你好，请问 arm cpu 的 mac 在配置docker-compose.yml机器学习、视频解码的硬件加速需要选择哪个？
 # extends:
    #   file: hwaccel.transcoding.yml
    #   service: cpu # set to one of [nvenc, quicksync, rkmpp, vaapi, vaapi-wsl] for accelerated transcoding

**Reply by sincerefly** [reply]
你好 👋，我没有修改这些配置，这几个都是借助显卡和核显加速的方式，看起来只能设置为 `cpu`

- nvenc（NVIDIA 显卡）
- quicksync（Intel 核显）
- rkmpp（Rockchip ARM 处理器）
- vaapi（AMD/Intel/NVIDIA (开源)）
- vaapi-wsl （	Intel/AMD (Windows)）

**Comment by Pasi-Cat** [comment]
大佬你好，请问immich的 机器学习 是如何运行起来的。

**Reply by sincerefly** [reply]
你好 👋，刚看到消息，服务部署后会自动运行，应该无需自行设置。建议参考官方文档部署，本文近一年时间未更新，价值有限。
